src/composables/useViewer.js

```diff
@@ -71,11 +73,26 @@ export function useViewer() {
 		// opened, so for the time being it is reparented if needed shortly
 		// after calling it.
 		// @see https://github.com/nextcloud/viewer/issues/995
-		setTimeout(() => {
-			if (store.getters.isFullscreen()) {
-				document.getElementById('content-vue').appendChild(document.getElementById('viewer'))
-			}
-		}, 1000)
+		watch(
+			() => store.getters.isFullscreen(),
```

Comment:
Is watching better, or would it make sense to emit an event in https://github.com/nextcloud/spreed/blob/47e98d86d617925253f69f4a5a7ad67d63fd4d79/src/store/windowVisibilityStore.js#L76-L78

Reply:
IMO, `watch` is better:
- This is a Vue way to react to a reactive state's change
- No need to manually remove a listener on clean-up, watch is managed by Vue + Handler is managed by Vue's scheduler
- Using an event emitter (a bus) for communication between Vue components is kind of an official anti-pattern in Vue (not for communication between different applications on the page). Here we have a direct and implicit dependency on isFullscreen state. With a global event, we have a star (as topologically)

Reply:
Roger